draft-ietf-dnsop-dns-capture-format.md

```diff
@@ -644,16 +601,18 @@ delay-pseconds | QR | Picosecond component of the time different between Query a
 ||
 query-name-index | QT | The index in the NAME/RDATA table of the QNAME for the first Question. Optional.
 ||
-query-size | R | The size of the DNS message (not the packet containing the message, just the DNS message) that forms the Query. Optional.
+query-size | R | DNS query message size (see below). Optional.
 ||
-response-size | R | The size of the DNS message (not the packet containing the message, just the DNS message) that forms the Response. Optional.
+response-size | R | DNS query message size (see below). Optional.
 ||
 query-extended | Q | Extended Query information. This item is only present if collection of extra Query information is configured. Optional.
 ||
 response-extended | R | Extended Response information. This item is only present if collection of extra Response information is configured. Optional.
 
 An implementation must always collects basic Q/R information. It may be configured to collect details on Question, Answer, Authority and Additional sections of the Query, the Response or both. Note that only the second and subsequent Questions of any Question section are collected (the details of the first are in the basic information), and that OPT Records are not collected in the Additional section.
 
+The query-size and response-size fields hold the DNS message size. For UDP this is the size of the UDP payload that contained the DNS message and will therefore include any trailing bytes if present. Trailing bytes with queries are routinely observed in traffic to authoritative servers and this value allows a calculation of how many trailing bytes were present. For TCP it is the size of the DNS message as specified in the two-byte message length header.
```

Comment:
I'm thinking of creating an issue on this for a future revision. The primary reason for adding the response field was to aid packet reconstruction in the face of name compression, and this in turn assumes that there is no trailing data in responses. I'm beginning to wonder if DNS message length AND transport length are required.